doc/DataPlaneCodeGeneration/Subclients.md

```diff
@@ -0,0 +1,295 @@
+# Customizing the generated code
+
+## Subclients
+
+There are two categories of clients: service clients and their subclients. Service clients can be instantiated and have the Client suffix. Subclients can only be created by calling factory methods on other clients (commonly on service clients) and do not have the client suffix. The service client is the entry point to the API for an Azure service. Please refer to [guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients) for more detail information.
+
+There are two tools to organize subclients and service client.
+
+- **Single top-level client**: define one top-level service client with the name <service_name>Client, all other clients will be its subclient.
+
+- **Client hierarchy**: organize the client service and subclients via defining the parent-subclient hierarchy dependency.
+
+**Note**: You can combine those two approaches to define an client hierarchy with one top-level client and its subclient may have sub clients also.
+
+In Following chapters, we will use [purview Account](https://github.com/Azure/azure-rest-api-specs/blob/b2bddfe2e59b5b14e559e0433b6e6d057bcff95d/specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2019-11-01-preview/account.json) as example.
+
+### Single top-level client
+
+With multiple service clients it may be better to group them all under one top-level client.
+
+<details>
+
+**Generated code before:**
+
+``` C#
+//Generated\AccountsClient.cs
+namespace Azure.Analytics.Purview.Account
+{
+    public partial class AccountsClient
+    {
+        public AccountsClient(string endpoint, TokenCredential credential, PurviewAccountClientOptions options = null){}
+    }
+}
+
+//Generated\CollectionsClient.cs
+namespace Azure.Analytics.Purview.Account
+{
+    public partial class CollectionsClient
+    {
+        public CollectionsClient(string endpoint, TokenCredential credential, PurviewAccountClientOptions options = null){}
+    }
+}
+
+//Generated\ResourceSetRulesClient.cs
+namespace Azure.Analytics.Purview.Account
+{
+    public partial class ResourceSetRulesClient
+    {
+        public ResourceSetRulesClient(string endpoint, TokenCredential credential, PurviewAccountClientOptions options = null){}
+    }
+}
+```
+
+**Add customize configuration (set single-top-level-client):**
+
+``` md
+### AutoRest Configuration
+> see https://aka.ms/autorest
+
+``` yaml
+namespace: Azure.Service.SubClients
+input-file: $(this-folder)/SubClients.json
+data-plane: true
+security: AzureKey
+security-header-name: Fake-Subscription-Key
+single-top-level-client: true
```

Comment:
Seems this is the key setting that makes the top level client. Is there a way to highlight that?

Reply:
I have not find a way to highlight in ymal code snippet in a md file.

Reply:
Then could you highlight that in the description under 
**Add customize configuration (set single-top-level-client):**